src/UnknownScannerException.java

```diff
@@ -29,7 +29,7 @@
 /**
  * Exception thrown when we try to use an invalid or expired scanner ID.
  */
-public final class UnknownScannerException extends RecoverableException
```

Comment:
This would require some more work as the scanner would need to be reopened with a new UID. What does it solve for you?